x-pack/plugins/security_solution/public/detections/components/rules/mitre/technique_fields.tsx

```diff
@@ -0,0 +1,212 @@
+import { MyAddItemButton } from '../add_item_form';
+import { getMitreErrorMessages, hasSubtechniqueOptions, isMitreTechniqueInvalid } from './helpers';
+import * as i18n from './translations';
+import { MitreSubtechniqueFields } from './subtechnique_fields';
+
+const TechniqueContainer = styled.div`
+  ${({ theme }) => css`
+    margin-left: 24px;
+    padding-left: 24px;
+    border-left: 2px solid ${theme.eui.euiColorLightestShade};
+  `}
+`;
+
+interface AddTechniqueProps {
+  field: FieldHook;
+  threatIndex: number;
+  idAria: string;
+  isDisabled: boolean;
+  onFieldChange: (threats: IMitreEnterpriseAttack[]) => void;
+}
+
+export const MitreTechniqueFields: React.FC<AddTechniqueProps> = ({
+  field,
+  idAria,
+  isDisabled,
+  threatIndex,
+  onFieldChange,
+}): JSX.Element => {
+  const [showValidation, setShowValidation] = useState(false);
+
+  const errorMessage = useMemo(() => {
+    const { techniqueError } = getMitreErrorMessages(field);
+    return techniqueError;
+  }, [field]);
+
+  const values = field.value as IMitreEnterpriseAttack[];
+
+  const removeTechnique = useCallback(
+    (index: number) => {
+      const threats = [...(field.value as IMitreEnterpriseAttack[])];
+      const techniques = threats[threatIndex].technique;
+      const newTechniques = [...techniques.slice(0, index), ...techniques.slice(index + 1)];
+      threats[threatIndex] = {
+        ...threats[threatIndex],
+        technique: newTechniques,
+      };
+      onFieldChange(threats);
+    },
+    [field, threatIndex, onFieldChange]
+  );
+
+  const addMitreTechnique = useCallback(() => {
+    const threats = [...(field.value as IMitreEnterpriseAttack[])];
+    threats[threatIndex] = {
+      ...threats[threatIndex],
+      technique: [
+        ...threats[threatIndex].technique,
+        { id: 'none', name: 'none', reference: 'none', subtechnique: [] },
+      ],
+    };
+    onFieldChange(threats);
+  }, [field, threatIndex, onFieldChange]);
+
+  const updateTechnique = useCallback(
+    (index: number, value: string) => {
+      const threats = [...(field.value as IMitreEnterpriseAttack[])];
+      const { id, reference, name } = techniquesOptions.find((t) => t.value === value) || {
+        id: '',
+        name: '',
+        reference: '',
+      };
+      onFieldChange([
+        ...threats.slice(0, threatIndex),
+        {
+          ...threats[threatIndex],
+          technique: [
+            ...threats[threatIndex].technique.slice(0, index),
+            {
+              id,
+              reference,
+              name,
+              subtechnique: [],
+            },
+            ...threats[threatIndex].technique.slice(index + 1),
+          ],
+        },
+        ...threats.slice(threatIndex + 1),
+      ]);
+    },
+    [threatIndex, onFieldChange, field]
+  );
+
+  const isTechniqueInvalid = useCallback((tacticName: string, technique: IMitreAttackTechnique) => {
+    return isMitreTechniqueInvalid(tacticName, technique);
```

Comment:
Just wanted to point out that we perform this validation at the UI level but not at the API level. This means that when you export a rule that was created prior to 7.11 and then import it into a 7.11 instance you could end up with a UI that looks like this. You are then able to save the rule as is from the Edit screen, even though it's technically "invalid".

![image](https://user-images.githubusercontent.com/616158/99792962-eb6bd980-2af5-11eb-906b-d9f91b657a27.png)